src/MongoDB.Driver.Core/Core/WireProtocol/Messages/Encoders/IMessageEncoderFactory.cs

```diff
@@ -90,5 +86,11 @@ public interface IMessageEncoderFactory
         /// </summary>
         /// <returns>An encoder.</returns>
         IMessageEncoder GetUpdateMessageEncoder();
+
+        /// <summary>
+        /// Gets an encoder for a Compressed message.
+        /// </summary>
+        /// <returns>An encoder.</returns>
+        IMessageEncoder GetCompressedMessageEncoder();
```

Comment:
Keep in alphabetical order?

Reply:
Also, should be:

        IMessageEncoder GetCompressedMessageEncoder(IMessageEncoderSelector originalEncoderSelector);

Reply:
done